File: Python/src/dss/collections/stack.py

class Stack:
  def __init__(self) -> None:
    self.__stack = []

  def __len__(self):
    return len(self.__stack)

  def peek(self):
    return self.__stack[len(self.__stack) - 1]

  def pop(self):
    return self.__stack.pop()  # this is cheating

  def push(self, value):
    self.__stack.append(value)  # also cheating

File: NodeJS/src/DataStructures/collections/Stack.js
export { Stack };

/**
 * @description: An array based stack implementation
 */

class Stack {
  #CONTAINER;
  constructor() {
    this.#CONTAINER = [];
  }

  /** Access */
  peek() {
    return this.#CONTAINER.at(-1);
  }

  /** Insertions */
  push(item) {
    this.#CONTAINER.push(item);
    return true;
  }

  /** Deletions */
  pop() {
    return this.#CONTAINER.pop();
  }

  /** Utility */
  isEmpty() {
    return this.#CONTAINER.length === 0;
  }
}
